src/lib/FileNode.cpp

```diff
@@ -23,110 +23,158 @@
-  case fnd_id::ObjectSpaceManifestListReferenceFND:
-  case fnd_id::ObjectSpaceManifestRootFND:
-  case fnd_id::FileDataStoreListReferenceFND:
-  case fnd_id::ObjectGroupStartFND:
-  case fnd_id::GlobalIdTableStart2FND:
-  case fnd_id::GlobalIdTableEntryFNDX:
-  case fnd_id::GlobalIdTableEndFNDX:
-  case fnd_id::DataSignatureGroupDefinitionFND:
-  case fnd_id::ObjectDeclaration2RefCountFND:
-  case fnd_id::ObjectGroupEndFND:
-  case fnd_id::ObjectInfoDependencyOverridesFND:
-  case fnd_id::RootObjectReference3FND:
-  case fnd_id::RevisionManifestEndFND:
-  case fnd_id::ObjectDeclarationFileData3RefCountFND:
-  case fnd_id::ReadOnlyObjectDeclaration2RefCountFND:
-  case fnd_id::FileDataStoreObjectReferenceFND:
-    break;
-  case fnd_id::fnd_invalid_id:
+    break;
+  case FndId::DataSignatureGroupDefinitionFND:
+    break;
+  case FndId::FileDataStoreListReferenceFND:
+    break;
+  case FndId::FileDataStoreObjectReferenceFND:
+    break;
+  case FndId::GlobalIdTableEntry2FNDX:
+    break;
+  case FndId::GlobalIdTableEntry3FNDX:
+    break;
+  case FndId::GlobalIdTableEntryFNDX:
+    break;
+  case FndId::GlobalIdTableStartFNDX:
+    break;
+  case FndId::HashedChunkDescriptor2FND:
+    break;
+  case FndId::ObjectDataEncryptionKeyV2FNDX:
+    break;
+  case FndId::ObjectDeclaration2LargeRefCountFND:
+    break;
+  case FndId::ObjectDeclaration2RefCountFND:
+    break;
+  case FndId::ObjectDeclarationFileData3LargeRefCountFND:
+    break;
+  case FndId::ObjectDeclarationFileData3RefCountFND:
+    break;
+  case FndId::ObjectDeclarationWithRefCount2FNDX:
+    break;
+  case FndId::ObjectDeclarationWithRefCountFNDX:
+    break;
+  case FndId::ObjectGroupListReferenceFND:
+    break;
+  case FndId::ObjectGroupStartFND:
+    break;
+  case FndId::ObjectInfoDependencyOverridesFND:
+    break;
+  case FndId::ObjectRevisionWithRefCount2FNDX:
+    break;
+  case FndId::ObjectRevisionWithRefCountFNDX:
+    break;
+  case FndId::ObjectSpaceManifestListReferenceFND:
+    break;
+  case FndId::ObjectSpaceManifestListStartFND:
+    break;
+  case FndId::ObjectSpaceManifestRootFND:
+    break;
+  case FndId::ReadOnlyObjectDeclaration2LargeRefCountFND:
+    break;
+  case FndId::ReadOnlyObjectDeclaration2RefCountFND:
+    break;
+  case FndId::RevisionManifestListReferenceFND:
+    break;
+  case FndId::RevisionManifestListStartFND:
+    break;
+  case FndId::RevisionManifestStart4FND:
+    break;
+  case FndId::RevisionManifestStart6FND:
+    break;
+  case FndId::RevisionManifestStart7FND:
+    break;
+  case FndId::RevisionRoleAndContextDeclarationFND:
+    break;
+  case FndId::RevisionRoleDeclarationFND:
+    break;
+  case FndId::RootObjectReference2FNDX:
+    break;
+  case FndId::RootObjectReference3FND:
+    break;
+
+  // nodes without data
+  case FndId::RevisionManifestEndFND:
+  case FndId::GlobalIdTableStart2FND:
+  case FndId::GlobalIdTableEndFNDX:
+  case FndId::ObjectGroupEndFND:
+  case FndId::ChunkTerminatorFND:
```

Comment:
For the sake of having a non-breaking change: the above file nodes are all valid, but would fall-through to `assert(false)`. I'd add an additional `break;` here